tests/generation/test_generation_utils.py

```diff
@@ -1336,6 +1405,64 @@ def test_constrained_beam_search_generate_dict_output(self):
             for output in (output_beam_search, output_generate):
                 self._check_outputs(output, input_ids, model.config, num_return_sequences=beam_scorer.num_beams)
 
+    def test_contrastive_generate(self):
+        # check `generate()` and `contrastive_search()` are equal
+        for model_class in self.all_generative_model_classes:
+
+            # TODO: fix me. Bloom fails because `past` has a different shape, while FSMT and reformer have a different
+            # cache variable type (and format).
+            if any(model_name in model_class.__name__.lower() for model_name in ["bloom", "fsmt", "reformer"]):
+                return
```

Comment:
Given the date of addition and usage rate for `fsmt` and `reformer`, I'm assuming we won't fix them -- is this correct?

Reply:
Yes, no need to make an effort for those two.